main.tf

```
@@ -85,7 +96,9 @@ locals {
 
   labels = [for l in local.label_order : local.id_context[l] if length(local.id_context[l]) > 0]
 
-  id_full = lower(join(local.delimiter, local.labels))
+  id_full = local.id_case == "lower" ? lower(join(local.delimiter, local.labels)) : (
+    local.id_case == "title" ? title(join(local.delimiter, local.labels)) :
+  upper(join(local.delimiter, local.labels)))
```

Add format type of "none", apply "title" case label by label, default to "lower"
```suggestion
  formatted_labels = [
    for l in local.labels : local.id_case == "none" ? l : local.id_case == "title" ? title(l) : local.id_case = "upper" ? upper(l) : lower(l)
  ]
      
  id_full = join(local.delimiter, local.formatted_labels)
```